honeybee_energy/material/glazing.py

```diff
@@ -0,0 +1,439 @@
+# coding=utf-8
+"""Glazing materials representing panes of glass within window constructions.
+
+They can exist anywhere within a window construction as long as they are not adjacent
+to other glazing materials.
+The one exception to this is the EnergyWindowMaterialSimpleGlazSys, which is meant to
+represent an entire window assembly (including glazing, gaps, and frame), and
+therefore must be the only material in its parent construction.
+"""
+from __future__ import division
+
+from ._base import _EnergyMaterialWindowBase
+from honeybee.typing import float_in_range, float_positive
+
+
+class _EnergyWindowMaterialGlazingBase(_EnergyMaterialWindowBase):
+    """Base for all glazing layers."""
+    _is_glazing_material = True
+
+
+class EnergyWindowMaterialGlazing(_EnergyWindowMaterialGlazingBase):
+    """A single glass pane corresponding to a layer in a window construction.
+
+    Properties:
+        name
+        thickness
+        solar_transmittance
+        solar_reflectance
+        solar_reflectance_back
+        visible_transmittance
+        visible_reflectance
+        visible_reflectance_back
+        infrared_transmittance
+        emissivity
+        emissivity_back
+        conductivity
+        dirt_correction
+        solar_diffusing
+        u_value
+        r_value
+    """
+    __slots__ = ('_name', '_thickness', '_solar_transmittance', '_solar_reflectance',
+                 '_solar_reflectance_back', '_visible_transmittance',
+                 '_visible_reflectance', '_visible_reflectance_back',
+                 '_infrared_transmittance', '_emissivity', '_emissivity_back',
+                 '_conductivity', '_dirt_correction', '_dirt_correction',
+                 '_solar_diffusing')
+    _thick_descr = 'window material thickness'
```

Comment:
Not a big fan of them being properties. I know it might be more typing but they should be kept inside methods if you ask me. It can also be confusing for other developers on what these attributes are and why are they exposed.

Reply:
Ok. If the answer to part 2 of my question above is that it does not save memory, I'll get rid of all of them without questions asked.

Reply:
After thinking about this in light of the `__init__` defaults that you mentioned to me the other day, I am nearly positive that the defaults here get stored on the class and not the instance, meaning what I did has no memory savings. Ok, I will keep the string defaults inside the methods.